test/unit/lib/validators/expected_results_validator_test.rb

```diff
@@ -60,6 +84,75 @@ def test_validate_missing_supplemental_data
+    @task.product_test.augmented_records = [@augmented_record1, @augmented_record3]
+    @record1.test_id = @task.product_test.id
+    @record1.save!
+    @record3.test_id = @task.product_test.id
+    @record3.save!
+    @validator.validate(file, 'task' => @task)
+    errors = @validator.errors
+    assert_equal 16, errors.length, 'should error on missing supplemental data' # 13 errors related to pop sums
+    errors.each { |e| (assert_equal :result_validation, e.validator_type) }
+    assert_equal 13, errors.count { |e| !pop_sum_err_regex.match(e.message).nil? }
+    assert_equal 3, errors.count { |e| e.validator_type == :result_validation && e.message == 'supplemental data error' }
+  end
+
+  def test_validate_augmented_results_three_augmented_patients_with_opposing_values
+    file = File.new(File.join(Rails.root, 'test/fixtures/qrda/ep_test_qrda_cat3_missing_supplemental_extra_male.xml')).read
+    @task.product_test.augmented_records = [@augmented_record1, @augmented_record2, @augmented_record3]
+    @record1.test_id = @task.product_test.id
+    @record1.save!
+    @record2.test_id = @task.product_test.id
+    @record2.save!
+    @record3.test_id = @task.product_test.id
+    @record3.save!
+    @validator.validate(file, 'task' => @task)
+    errors = @validator.errors
+    assert_equal 16, errors.length, 'should error on missing supplemental data' # 13 errors related to pop sums
+    errors.each { |e| (assert_equal :result_validation, e.validator_type) }
+    assert_equal 13, errors.count { |e| !pop_sum_err_regex.match(e.message).nil? }
+    assert_equal 3, errors.count { |e| e.validator_type == :result_validation && e.message == 'supplemental data error' }
+  end
+
+  def test_validate_augmented_results_three_augmented_patients_reporting_extra_male
+    file = File.new(File.join(Rails.root, 'test/fixtures/qrda/ep_test_qrda_cat3_missing_supplemental_two_extra_male.xml')).read
+    @task.product_test.augmented_records = [@augmented_record1, @augmented_record2, @augmented_record3]
+    @record1.test_id = @task.product_test.id
+    @record1.save!
+    @record2.test_id = @task.product_test.id
+    @record2.save!
+    @record3.test_id = @task.product_test.id
+    @record3.save!
+    @validator.validate(file, 'task' => @task)
+    errors = @validator.errors
+
+    error_details = { type: 'supplemental_data',
+                      population_key: 'IPP',
+                      population_id: 'E83A6DA6-D34E-4107-8431-2FB2C86738C7',
+                      data_type: 'SEX',
+                      code: 'F',
+                      expected_value: 3,
+                      reported_value: 1 }
+
+    # The should be error messages when reported value is outside of the expected range.  In this example, the range is 2-3.
+    assert_equal 3, errors.count { |e| e.validator_type == :result_validation && e.error_details == error_details }
+  end
+
```

Comment:
There should be one last test case for a good result due to the augmented patients.

Reply:
Actually, scratch that. These are meant to be "good" cases. If they could have a comment added to them mentioning that, since they read like "bad" cases since they expect errors--they just expect no more errors than before.